test/toolkits/style_transfer/test_style_transfer_model_definition.cxx

```diff
@@ -0,0 +1,186 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#define BOOST_TEST_MODULE
+
+#include <map>
+#include <memory>
+#include <set>
+#include <string>
+
+#include <boost/asio.hpp>
+#include <boost/test/unit_test.hpp>
+#include <core/util/test_macros.hpp>
+#include <ml/neural_net/float_array.hpp>
+#include <toolkits/coreml_export/mlmodel_include.hpp>
+#include <toolkits/style_transfer/style_transfer_model_definition.hpp>
+
+#include "utils.hpp"
+
+using CoreML::Specification::InnerProductLayerParams;
+using CoreML::Specification::NeuralNetwork;
+using CoreML::Specification::NeuralNetworkLayer;
+
+using namespace turi::style_transfer;
+using namespace turi::neural_net;
+
+#define RESNET_MODEL                                           \
+  "https://docs-assets.developer.apple.com/turicreate/models/" \
+  "transformer.mlmodel"
+#define VGG_16_MODEL \
+  "https://docs-assets.developer.apple.com/turicreate/models/vgg16.mlmodel"
```

Comment:
Why do we need to download the actual models to test this code? I would prefer it if our unit tests have few dependencies, and we can reserve this kind of thing for end-to-end integration tests

Reply:
Fair, I didn't know how to test the functionality of the code without the actual MLModels. For VGG_16 can't test the network without the MLModel. For Resnet 16 can only test the Network Architecture. What do you suggest I do here?

Reply:
Defined the spec and exported to coreML in the utils file.